test/app.test.js

```diff
@@ -91,15 +91,83 @@ describe('app', () => {
     db.query = () => Promise.reject('This is a test of the emergency broadcast system.');
 
     const dataRequest = request(app)
-      .get('/v1/agencies/fake-agency/reports/fake-report/data')
+      .get('/v1.1/agencies/fake-agency/reports/fake-report/data')
       .expect(400);
 
     dataRequest.then(response => {
       expect(response.body).to.deep.equal({
-        message: 'An error occured. Please check the application logs.',
+        message: 'An error occurred. Please check the application logs.',
         status: 400
       });
       done();
     }).catch(done);
   });
+
+  it('should respond with a 400 if the domain report is not one of the acceptable kinds of reports', done => {
+    db.query = (params) => {
+      expect(params.domain).to.equal('fakeiscool.gov');
+      expect(params.reportName).to.equal('browser');
+      return Promise.resolve([
+        { id: 1, date: new Date('2017-01-01'), data: { domain: 'fakeiscool.gov' } },
+        { id: 2, date: new Date('2017-01-02'), data: { domain: 'bobtown.gov' } }
+      ]);
+    };
+
+    const dataRequest = request(app)
+      .get('/v1.1/domain/fakeiscool.gov/reports/browser/data')
+      .expect(400);
+
+    dataRequest.then(response => {
+      expect(response.body).to.deep.equal({
+        message: 'You are requesting a report that cannot be filtered on domain. Please try one of the following reports: site, domain, download, second-level-domain.',
+        status: 400
+      });
+      done();
+    }).catch(done);
+  });
+
+  it('should pass params from the url to db.query and render the result for a domain query for a non-download report', done => {
+    db.query = (params) => {
+      expect(params.domain).to.equal('fakeiscool.gov');
+      expect(params.reportName).to.equal('site');
+      return Promise.resolve([
+        { id: 1, report_name: 'site', date: new Date('2017-01-01'), data: { domain: 'fakeiscool.gov' } }
+      ]);
+    };
+
+    const dataRequest = request(app)
+      .get('/v1.1/domain/fakeiscool.gov/reports/site/data')
+      .expect(200);
+
+    dataRequest.then(response => {
+      expect(response.body).to.deep.equal([
+        { id: 1, date: '2017-01-01', domain: 'fakeiscool.gov' }
```

Comment:
Looks like the code is working, but the test is acting up. I believe we need to add a `report_name` prop in here.

Reply:
ah thanks, looks like this hadn't been pushed.